### PR TITLE
Redesign modifications with groovy script and switch state modifications

### PR DESCRIPTION
**Pull request by FranckLecuyer**
PR must be merged after PR : https://github.com/gridsuite/network-modification-server/pull/61

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java [hunk 1]
```diff
@@ -236,8 +246,8 @@ public void onRemoval(Identifiable identifiable) {
         //equipmentInfosService.delete(identifiable.getId(), networkUuid);
     }
 
-    public void setSubstationsIds(Set<String> substationsIds) {
-        this.substationsIds = substationsIds;
+    public void addSubstationsIds(Set<String> substationsIds) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:249]: Remove it and add VLs in the `onCreation` and the `deleteEquipmentInfos` methods
and rename `deleteEquipmentInfos` by `onTemporaryRemoval`

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/dto/GroovyScriptModificationInfos.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "Groovy script modification")
+public class GroovyScriptModificationInfos extends ModificationInfos {
+    @Schema(description = "Groovy script")
+    private String script;
+
+    @Schema(description = "Substations ID")
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/dto/GroovyScriptModificationInfos.java:32]: Done

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 3]
```diff
@@ -213,6 +214,19 @@ public EquipmentDeletionEntity createEquipmentDeletionEntity(String equipmentId,
         return new EquipmentDeletionEntity(equipmentId, equipmentType);
     }
 
+    public GroovyScriptModificationEntity createGroovyScriptModificationEntity(String script) {
+        return new GroovyScriptModificationEntity(script);
+    }
+
+    public GroovyScriptModificationInfos getGroovyScriptModification(UUID groupUuid, UUID modificationUuid) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Removed

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 4]
```diff
@@ -155,8 +165,22 @@ public NetworkModificationService(@Value("${backing-services.report-server.base-
                 if (aSwitch.isOpen() != open) {
                     aSwitch.setOpen(open);
                 }
+
+                // store the substations ids in the listener
+                listener.addSubstationsIds(NetworkStoreListener.getSubstationIds(aSwitch));
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:170]: Removed

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java [hunk 5]
```diff
@@ -34,7 +33,7 @@
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:33]: Removed

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java [hunk 6]
```diff
@@ -179,6 +185,10 @@ public void storeTwoWindingsTransformerCreation(TwoWindingsTransformerCreationIn
         );
     }
 
+    public void storeGroovyScriptModification(String script) {
+        modifications.add(this.modificationRepository.createGroovyScriptModificationEntity(script));
+    }
+
     private Set<String> getSubstationsIds(String equipmentId) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:192]: Removed

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 7]
```diff
@@ -52,7 +52,7 @@ public NetworkModificationController(NetworkModificationService networkModificat
     @PutMapping(value = "/networks/{networkUuid}/groovy", consumes = MediaType.TEXT_PLAIN_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     @Operation(summary = "change an equipment state in a network variant")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The equipment state has been changed")})
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:52]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 8]
```diff
@@ -52,7 +52,7 @@ public NetworkModificationController(NetworkModificationService networkModificat
...
-    public ResponseEntity<Flux<EquipmenModificationInfos>> applyGroovyScript(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
+    public ResponseEntity<Flux<ModificationInfos>> applyGroovyScript(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:53]: study-server only have to know the substations ids involved in the modification ...

File: src/main/java/org/gridsuite/modification/server/entities/GroovyScriptModificationEntity.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    @Column(name = "script", columnDefinition = "CLOB")
+    private String script;
+
+    public GroovyScriptModificationEntity(String script) {
+        super(ModificationType.GROOVY_SCRIPT);
+        this.script = script;
+    }
+
+    @Override
+    public GroovyScriptModificationInfos toModificationInfos() {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/GroovyScriptModificationEntity.java:39]: Removed

File: src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java [hunk 10]
```diff
@@ -185,11 +185,11 @@ public void testLoadCreation() {
         assertEquals(3, modificationInfos.size());
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java:185]: Done

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Used in ModificationRepositoryTest now

File: src/main/java/org/gridsuite/modification/server/entities/GroovyScriptModificationEntity.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/GroovyScriptModificationEntity.java:39]: Used in ModificationRepositoryTest now

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Better to move in the class test ?
Like testStatusLineModification in https://github.com/gridsuite/network-modification-server/pull/67

File: src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/service/NetworkStoreListener.java:241]: Done in https://github.com/gridsuite/network-modification-server/pull/67

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Better to move in the class test ?
Like testStatusLineModification in https://github.com/gridsuite/network-modification-server/pull/67

[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Better to move in the class test ?
Like testStatusLineModification in https://github.com/gridsuite/network-modification-server/pull/67

[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:221]: Better to move in the class test ?
Like testStatusLineModification in #67

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**